core/operation/src/main/java/uk/gov/gchq/gaffer/operation/impl/Repeat.java

```diff
@@ -0,0 +1,133 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.gov.gchq.gaffer.operation.impl;
+
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import com.fasterxml.jackson.core.type.TypeReference;
+import org.apache.commons.lang3.exception.CloneFailedException;
+
+import uk.gov.gchq.gaffer.operation.Operation;
+import uk.gov.gchq.gaffer.operation.Operations;
+import uk.gov.gchq.gaffer.operation.io.InputOutput;
+import uk.gov.gchq.gaffer.operation.serialisation.TypeReferenceImpl;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * A {@code Repeat} is a Gaffer {@link Operation} for executing a delegate <code>Operation</code>
+ * for a specified number of repeats.
+ * <p>The delegate operation can also be an implementation of {@link Operations},
+ * so for example, a small {@link uk.gov.gchq.gaffer.operation.OperationChain} could be delegated
+ * to the <code>Repeat</code>, and repeated a number of times.</p>
+ */
+@JsonPropertyOrder(value = {"class", "input", "operation", "times"}, alphabetic = true)
+public class Repeat implements InputOutput<Object, Object>,
+        Operations<Operation> {
+    private Operation operation;
+    private int times;
+    private Object input;
+    private Map<String, String> options;
+
+    public Repeat() {
+
+    }
+
+    public Repeat(final Operation operation, final int times) {
+        this.operation = operation;
+        this.times = times;
+    }
+
+    public void setOperation(final Operation operation) {
+        this.operation = operation;
+    }
+
+    public Operation getOperation() {
+        return operation;
+    }
+
+    public void setTimes(final int times) {
+        this.times = times;
+    }
+
+    public int getTimes() {
+        return times;
+    }
+
+    @Override
+    public Object getInput() {
+        return input;
+    }
+
+    @Override
+    public void setInput(final Object input) {
+        this.input = input;
+    }
+
+    @Override
+    public TypeReference<Object> getOutputTypeReference() {
+        return new TypeReferenceImpl.Object();
+    }
+
+    @Override
+    public Repeat shallowClone() throws CloneFailedException {
+        return new Repeat.Builder()
+                .input(input)
+                .operation(operation)
+                .times(times)
+                .options(options)
+                .build();
+    }
+
+    @Override
+    public Map<String, String> getOptions() {
+        return options;
+    }
+
+    @Override
+    public void setOptions(final Map<String, String> options) {
+        this.options = options;
+    }
+
+    @Override
+    public Collection<Operation> getOperations() {
+        final List<Operation> ops = new ArrayList<>();
+        for (int i = 0; i < times; i++) {
+            ops.add(operation);
+        }
+        return ops;
+    }
```

Comment:
You will also need to implement updateOperations. This should have some logic like:
if(operations.size() > 1) {
operation = new OperationChain(operations);
} else {
operation = operations.get(0);
}